xbmc/cores/AudioEngine/Sinks/AESinkDirectSound.cpp

```diff
@@ -119,56 +104,42 @@ std::unique_ptr<IAESink> CAESinkDirectSound::Create(std::string& device,
   return {};
 }
 
-bool CAESinkDirectSound::Initialize(AEAudioFormat &format, std::string &device)
+bool CAESinkDirectSound::Initialize(AEAudioFormat& format, std::string& device)
 {
   if (m_initialized)
     return false;
 
   bool deviceIsBluetooth = false;
-  LPGUID deviceGUID = nullptr;
-  RPC_WSTR wszUuid  = nullptr;
+  GUID deviceGUID = {};
   HRESULT hr = E_FAIL;
   std::string strDeviceGUID = device;
-  std::list<DSDevice> DSDeviceList;
-  std::string deviceFriendlyName;
-  DirectSoundEnumerate(DSEnumCallback, &DSDeviceList);
-
-  if(StringUtils::EndsWithNoCase(device, std::string("default")))
-    strDeviceGUID = GetDefaultDevice();
+  std::string deviceFriendlyName = "unknown";
+  
+  if (StringUtils::EndsWithNoCase(device, "default"))
+    strDeviceGUID = CAESinkFactoryWin::GetDefaultDeviceId();
 
-  for (std::list<DSDevice>::iterator itt = DSDeviceList.begin(); itt != DSDeviceList.end(); ++itt)
-  {
-    if ((*itt).lpGuid)
-    {
-      hr = (UuidToString((*itt).lpGuid, &wszUuid));
-      std::string sztmp = KODI::PLATFORM::WINDOWS::FromW(reinterpret_cast<wchar_t*>(wszUuid));
-      std::string szGUID = "{" + std::string(sztmp.begin(), sztmp.end()) + "}";
-      if (StringUtils::CompareNoCase(szGUID, strDeviceGUID) == 0)
-      {
-        deviceGUID = (*itt).lpGuid;
-        deviceFriendlyName = (*itt).name.c_str();
-        break;
-      }
-    }
-    if (hr == RPC_S_OK) RpcStringFree(&wszUuid);
```

Comment:
Now `#include <Rpc.h>` and `#pragma comment(lib, "Rpcrt4.lib")` lines can be removed.

Reply:
right! and a couple more actually.